components/member-profile/index.js

```diff
@@ -1,29 +1,29 @@
-import { getDataURL, getImgURL } from 'helper-functions/urls';
-
 import PropTypes from 'prop-types';
 import React from 'react';
 import classNames from './member-profile.module.scss';
-import useFetch from 'custom-hooks/useFetch';
 
 const Profile = (props) => {
-  const imgLink = getImgURL(props.id);
-  const { data } = useFetch(getDataURL(props.id));
+  const memberName = `${props.membersData.first_name || '--'} ${
+    props.membersData.last_name || '--'
+  }`;
   return (
     <>
-      <img src={imgLink} className={classNames.profilePic} alt="profile" />
-      <div className={classNames.memberName}>
-        {data ? `${data['first_name']} ${data['last_name']} ` : `${props.id}`}
-      </div>
+      <img src={props.imageLink} className={classNames.profilePic} alt="profile" />
+      <div className={classNames.memberName}>{memberName}</div>
     </>
   );
 };
 
 Profile.propTypes = {
-  id: PropTypes.string
+  imageLink: PropTypes.string
+};
+
+Profile.propTypes = {
+  membersData: PropTypes.object
 };
```

Comment:
Why are here two propTypes? Both `membersData` and `imageLink` can be defined inside one propTypes object.

Reply:
if you know the prop structure, try to be specific in `propTypes`.

For example,

```
Profile.propTypes={
  membersData:PropTypes. shape({
        first_name: PropTypes.string,
        last_name: PropTypes.string
  })
}
```

If not required, default it too.

Reply:
@sumitd94 Please follow @shihabus's recommendation

Reply:
Did not understand this part.. could you please explain more on this.

Reply:
@Shihab-Duzno Does this mean we are defining the properties of object (in this case membersData) more clearly ?

Reply:
@sumitd94 yes. It is to define the expected structure and type of data. If the data type is something different we can get warnings too. Some kind of type checking.

When default the value also, keep the same structure.

Reply:
Defaulting like this : 

Profile.defaultProps = {
  imageLink: '',
  membersData: PropTypes.shape({
    first_name: 'Sumit',
    last_name: 'Sumit'
  })
};

Is this the right way ? @Shihab-Duzno

Reply:
```
Profile.defaultProps = {
  imageLink: '',
  membersData:{
    first_name: '',
    last_name: ''
  }
};
```

Reply:
Ohh I see, Got it
Thank you :+1: